src/components/retrieve_from_weaviate/Dockerfile

```diff
@@ -0,0 +1,22 @@
+FROM --platform=linux/amd64 pytorch/pytorch:2.0.1-cuda11.7-cudnn8-runtime
+# System dependencies
+RUN apt-get update && \
+    apt-get upgrade -y && \
+    apt-get install git -y
+
+# Install requirements
+COPY requirements.txt ./
+RUN pip3 install --no-cache-dir -r requirements.txt
+
+# Install Fondant
+# This is split from other requirements to leverage caching
+ARG FONDANT_VERSION=main
```

Comment:
This is the Dockerfile for reusable components in the Fondant repository. If we're adding it in this repo, I would use the custom Dockerfile instead (similar to the one you used for the csv reader). If we move the components to the Fondant repo later, we can update the Dockerfiles.